src/query/range_query/range_query.rs

```diff
@@ -364,7 +385,9 @@ impl Weight for RangeWeight {
         let inverted_index = reader.inverted_index(reader.schema().get_field(&self.field)?)?;
         let term_dict = inverted_index.terms();
         let mut term_range = self.term_range(term_dict)?;
-        while term_range.advance() {
+        let mut processed_count = 0;
+        while term_range.advance() && self.limit.map_or(true, |limit| processed_count < limit) {
```

Comment:
nitpick: (meaning no need to fix) `map_or` is not helping readability.

```rust
if let Some(limit) = self.limit {
    if processed_coutn < limit {
          break;
    }
}
```

within the body is my favorite solution on something like that.